src/components/dialog/faultEvent/FaultEventCreation.tsx

```diff
@@ -63,6 +63,11 @@ const FaultEventCreation = ({ useFormMethods, isRootEvent, eventValue, isEditedE
   };
 
   function renderEventSelect() {
+    const supertype =
```

Comment:
This too complicated, how about:
```
const supertype = eventValue?.supertypes?.[0] || null;
const eventVal = supertype || eventValue;
const defaultValue = eventVal ? { name: eventVal.name, iri: eventVal.iri } : null;
```

It can be even simpler if you do not need null value, but undefined instead

Reply:
or optional array chaining see https://stackoverflow.com/questions/59623674/how-can-i-use-optional-chaining-with-arrays-and-functions